portal-api/src/v1/controllers/banks.controller.js

```diff
@@ -55,3 +57,22 @@ exports.delete = async (req, res) => {
   const status = await collection.deleteOne({ id: req.params.id });
   res.status(200).send(status);
 };
+
+// validate the user's bank against the deal
+exports.validateBank = async (req, res) => {
+  const { dealId, bankId } = req.body;
+
+  // check if the `dealId` is a valid ObjectId
+  if (hasValidObjectId(dealId)) {
+    const collection = await db.getCollection('deals');
+
+    // validate the bank against the deal
+    const isValid = await collection.findOne({ _id: ObjectId(dealId), 'bank.id': bankId });
+
+    if (isValid) {
+      return res.status(200).send({ status: 200, isValid: true });
+    }
+    return res.status(404).send({ status: 404, isValid: false });
```

Comment:
I think 404 may be better here because it checks the bank against the deal - it already passed the validation process above so at this stage it wouldn't be a `bad request`

however, `422` could change to 400 because above I'm validating if an object is valid. and if it's not valid, then it's a `bad request`